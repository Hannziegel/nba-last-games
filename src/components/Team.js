import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';
import '../styles/Teams.css';
import { useDispatch } from 'react-redux';
import { IoArrowForwardCircle } from 'react-icons/io5';
import { getTeamLastGame } from '../redux/teamLastGames';

const Team = (props) => {
  const sinceText = 'Since ';
  const { team } = props;
  const dispatch = useDispatch();

  const handleClick = () => {
    dispatch(getTeamLastGame(team.idTeam));
  };

  return (
    <NavLink to={`/${team.strTeam}`} className="team-container" onClick={handleClick}>
      <IoArrowForwardCircle className="button-arrow-right" />
      <img className="team-badge-img" alt="team badge" src={team.strTeamBadge} />
      <h3 className="team-name">{team.strTeam}</h3>
      <span className="team-info">
        {sinceText}
        {team.intFormedYear}
      </span>
    </NavLink>
  );
};

Team.propTypes = {
  team: PropTypes.shape({
    idTeam: PropTypes.string,
    strTeam: PropTypes.string,
    intFormedYear: PropTypes.string,
    strTeamBadge: PropTypes.string,
  }).isRequired,
};

export default Team;
