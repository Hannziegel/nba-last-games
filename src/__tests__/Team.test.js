import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import React from 'react';
import store from '../Redux/configureStore';
import Team from '../components/Team';

describe('App js testing', () => {
  const team = {
    strHomeTeam: 'PropTypes.string',
    intHomeScore: 'PropTypes.string',
    strAwayTeam: 'PropTypes.string',
    intAwayScore: 'PropTypes.string',
    strThumb: 'PropTypes.string',
  };

  test('Renders App', () => {
    const tree = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <Team team={team} />
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(tree).toMatchSnapshot();
  });
});
