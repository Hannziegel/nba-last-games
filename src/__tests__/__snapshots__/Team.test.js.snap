// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App js testing Renders App 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <a
        class="team-container"
        href="/undefined"
      >
        <svg
          class="button-arrow-right"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 256c0-114.87-93.13-208-208-208S48 141.13 48 256s93.13 208 208 208 208-93.13 208-208zm-212.65 91.36a16 16 0 01-.09-22.63L303.58 272H170a16 16 0 010-32h133.58l-52.32-52.73A16 16 0 11274 164.73l79.39 80a16 16 0 010 22.54l-79.39 80a16 16 0 01-22.65.09z"
          />
        </svg>
        <img
          alt="team badge"
          class="team-badge-img"
        />
        <h3
          class="team-name"
        />
        <span
          class="team-info"
        >
          Since 
        </span>
      </a>
    </div>
  </body>,
  "container": <div>
    <a
      class="team-container"
      href="/undefined"
    >
      <svg
        class="button-arrow-right"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M464 256c0-114.87-93.13-208-208-208S48 141.13 48 256s93.13 208 208 208 208-93.13 208-208zm-212.65 91.36a16 16 0 01-.09-22.63L303.58 272H170a16 16 0 010-32h133.58l-52.32-52.73A16 16 0 11274 164.73l79.39 80a16 16 0 010 22.54l-79.39 80a16 16 0 01-22.65.09z"
        />
      </svg>
      <img
        alt="team badge"
        class="team-badge-img"
      />
      <h3
        class="team-name"
      />
      <span
        class="team-info"
      >
        Since 
      </span>
    </a>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
